api/access/server.go

```diff
@@ -0,0 +1,515 @@
+	}
+
+	var events []*access.EventsResponse_Result
+	for height := in.StartHeight; height <= in.EndHeight; height++ {
+		ee, err := s.index.Events(height, types...)
+		if err != nil {
+			return nil, fmt.Errorf("could not get events at height %d: %w", height, err)
+		}
+
+		header, err := s.index.Header(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get header at height %d: %w", height, err)
+		}
+
+		timestamp, err := ptypes.TimestampProto(header.Timestamp)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse timestamp for block at height %d: %w", height, err)
+		}
+
+		messages := make([]*entities.Event, 0, len(ee))
+		for _, event := range ee {
+			messages = append(messages, convert.EventToMessage(event))
+		}
+
+		blockID := header.ID()
+		result := access.EventsResponse_Result{
+			BlockId:        blockID[:],
+			BlockHeight:    height,
+			BlockTimestamp: timestamp,
+			Events:         messages,
+		}
+
+		events = append(events, &result)
+	}
+
+	resp := access.EventsResponse{
+		Results: events,
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetEventsForBlockIDs(_ context.Context, in *access.GetEventsForBlockIDsRequest) (*access.EventsResponse, error) {
+	var types []flow.EventType
+	if in.Type != "" {
+		types = append(types, flow.EventType(in.Type))
+	}
+
+	var events []*access.EventsResponse_Result
+	for _, id := range in.BlockIds {
+		blockID := flow.HashToID(id)
+		height, err := s.index.HeightForBlock(blockID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get height of block with ID %x: %w", id, err)
+		}
+
+		ee, err := s.index.Events(height, types...)
+		if err != nil {
+			return nil, fmt.Errorf("could not get events at height %d: %w", height, err)
+		}
+
+		header, err := s.index.Header(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get header at height %d: %w", height, err)
+		}
+
+		timestamp, err := ptypes.TimestampProto(header.Timestamp)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse timestamp for block at height %d: %w", height, err)
+		}
+
+		messages := make([]*entities.Event, 0, len(ee))
+		for _, event := range ee {
+			messages = append(messages, convert.EventToMessage(event))
+		}
+
+		result := access.EventsResponse_Result{
+			BlockId:        blockID[:],
+			BlockHeight:    height,
+			BlockTimestamp: timestamp,
+			Events:         messages,
+		}
+
+		events = append(events, &result)
+	}
+
+	resp := access.EventsResponse{
+		Results: events,
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetNetworkParameters(_ context.Context, _ *access.GetNetworkParametersRequest) (*access.GetNetworkParametersResponse, error) {
+	root, err := s.index.First()
+	if err != nil {
+		return nil, fmt.Errorf("could not get first indexed height: %w", err)
+	}
+
+	header, err := s.index.Header(root)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header: %w", err)
+	}
+
+	return &access.GetNetworkParametersResponse{ChainId: header.ChainID.String()}, nil
```

Comment:
I personally don't like in-lining it like this. My rationale is similar to having imports always with brackets, and always a trailing coma. It's more consistent to always return a variable, and it won't hurt performance. It adds clarity at no cost to explicitly initialize the response in its own block, with fields on their own line.